src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -711,9 +711,35 @@ private void closeTempFile(String encodedPartition) {
   }
 
   private void closeTempFile() {
+    ConnectException connectException = null;
     for (String encodedPartition : tempFiles.keySet()) {
       // Close the file and propagate any errors
-      closeTempFile(encodedPartition);
+      try {
+        closeTempFile(encodedPartition);
```

Comment:
what about the other call to `closeTempFile(encodedPartition)`? Do we risk to leave stale tmp files in that case too?

Reply:
I am not sure what you mean. This guards against both calls and yes the empty file situation could occur in both situations (if that's what you are asking)

Reply:
This method is called in `close` too

Reply:
Yes and if it fails to close the file, it will delete it as per normal. There is no risk of leaving stale tmp files.

Reply:
From what I can see, `closeTempFile` and `deleteTempFile` are called within the same [try/catch](https://github.com/confluentinc/kafka-connect-hdfs/blob/b76d0422240594d0b5b619f7ff2b3e5b5f45d6ea/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L449-L450) block and if `closeTempFile` throws an exception, `deleteTempFile` will be skipped. If it's called later on again, I think it's worth mentioning what the actual workflow is, because it's not obvious.

Reply:
yes I see what you mean now and yes that can happen and can leave stale temp files, but that logic is out of the scope of this PR. 

This PR is interested in the case when we try to close temp file so that we can commit it to HDFS and that operation fails. 

Closing the TPW gracefully to avoid stale temp files should be addressed in a different PR.

Reply:
Sorry, I missed the reply here. 

I don't think we are extending the scope too much if we add proper handling of the exception at that point too, given that we do that for this call here. Is there anything else that needs to happen besides deleting the file even in the presence of exception during close?

Reply:
@kkonstantine I have added graceful handling of exceptions in `close()`